### PR TITLE
feature #4183 - token transfer history gets actual transaction timest…

### LINKED ISSUE #4183: Token transaction history shows wrong date
[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a user, I want to see token transaction info so that I can know date/amount/sender/etc.

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Bug

[comment]: # (Describe the feature you would like, or briefly summarise the bug and what you did, what you expected to happen, and what actually happens. Sections below)
*Summary*: I received a 100 SNT token transfer 22 days ago, but Status is saying it was sent 1 minute ago. I verified when the transaction was sent by tapping more, then viewing on Etherscan

#### Expected behavior
Actual sent date is shown

#### Actual behavior
Todays date and current time are shown)

### Reproduction
Transfer SNT and wait.

### Additional Information
* Status version: Nightly
* Operating System: iOS

### PR DESCRIPTION
**Pull request by goranjovic**
fixes #4183

### Summary:

Fixes the issue with incorrect timestamps (always current time) in incoming token transactions by getting the actual timestamp from block info.

### Steps to test:
- Open Status
- Send a transaction from another phone 
- See it appear in the history and check the time
- Check history after some time to see if the time is still ok

status: ready

### CODE REVIEW

File: src/status_im/utils/ethereum/erc20.cljs [hunk 1]
```diff
@@ -105,7 +105,7 @@
               ;; NOTE(goranjovic) - timestamp is mocked to the current time so that the transaction is shown at the
```
Comment (jeluard) [inline src/status_im/utils/ethereum/erc20.cljs]: Should this comment be removed?

File: src/status_im/utils/ethereum/erc20.cljs [hunk 2]
```diff
@@ -140,7 +151,7 @@
 
 (defn get-token-transfer-logs
   ;; NOTE(goranjovic): here we cannot use web3 since events don't work with infura
```
Comment (jeluard) [inline src/status_im/utils/ethereum/erc20.cljs]: This comment is a little weird now that `web3` is used.

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/KtzCLZ for branch PR-4248

<blockquote><div><strong><a href="https://i.diawi.com/KtzCLZ">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/GotQBZ for branch PR-4248

<blockquote><div><strong><a href="https://i.diawi.com/GotQBZ">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

File: src/status_im/utils/ethereum/erc20.cljs (hunk 2 above)
Comment (goranjovic) [reply src/status_im/utils/ethereum/erc20.cljs]: clarified the comment

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/xM2NnV for branch PR-4248

<blockquote><div><strong><a href="https://i.diawi.com/xM2NnV">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/J4GUuY for branch PR-4248

<blockquote><div><strong><a href="https://i.diawi.com/J4GUuY">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Review by jeluard — APPROVED** [review]

**Review by flexsurfer — APPROVED** [review]

**Comment by asemiankevich** [comment]
@goranjovic i dont see SNT token transaction in my recovered account, however i do see SNT on my balance :(

**Comment by asemiankevich** [comment]
sorry nevermind  - it appeared but with a timelag

**Comment by goranjovic** [comment]
@asemiankevich Is its timestamp ok? Especially if it is a transfer where you received the SNT?

**Comment by asemiankevich** [comment]
@goranjovic it differs from etherscan , status shows May 8 12:21 AM and here is etherscan https://etherscan.io/tx/0xb1ba8ae3f16a69eb8c1dd43bfc2e9c0dd8b4ac7a1e6ea2ae7f6205d64a537fb5

<blockquote><div><strong><a href="https://etherscan.io/tx/0xb1ba8ae3f16a69eb8c1dd43bfc2e9c0dd8b4ac7a1e6ea2ae7f6205d64a537fb5">Ethereum Transaction 0xb1ba8ae3f16a69eb8c1dd43bfc2e9c0dd8b4ac7a1e6ea2ae7f6205d64a537fb5</a></strong></div><div>Ethereum (ETH) detailed transaction info for 0xb1ba8ae3f16a69eb8c1dd43bfc2e9c0dd8b4ac7a1e6ea2ae7f6205d64a537fb5</div></blockquote>

**Comment by goranjovic** [comment]
@asemiankevich That's weird. I'll need to take a look.
Can you post the screenshot of transaction details screen for this?

**Comment by asemiankevich** [comment]
image

**Comment by goranjovic** [comment]
@asemiankevich And just for completeness, what's your phone time zone?

**Comment by asemiankevich** [comment]
@goranjovic it is UTC+3 , do you think it is the case?

**Comment by goranjovic** [comment]
@asemiankevich 

May-07-2018 09:21:45 PM +UTC and add 3 hours and yep, that's it. 21 min after midnight, technically already 8th.

Yes, that's it - you get your local time in Status and in Etherscan it's always in UTC.

**Pull request merged**